src/vs/workbench/contrib/chat/browser/chatContentParts/chatToolInvocationPart.ts

```diff
@@ -273,7 +288,26 @@ class ChatToolInvocationSubPart extends Disposable {
 		}
 
 		this._register(confirmWidget.onDidClick(button => {
-			toolInvocation.confirmed.complete(button.data);
+			switch (button.data as ConfirmationOutcome) {
+				case ConfirmationOutcome.AllowGlobally:
+					this.languageModelToolsService.setToolAutoConfirmation(toolInvocation.toolId, 'profile', true);
+					toolInvocation.confirmed.complete(true);
+					break;
+				case ConfirmationOutcome.AllowWorkspace:
+					this.languageModelToolsService.setToolAutoConfirmation(toolInvocation.toolId, 'workspace', true);
+					toolInvocation.confirmed.complete(true);
+					break;
+				case ConfirmationOutcome.AllowSession:
+					this.languageModelToolsService.setToolAutoConfirmation(toolInvocation.toolId, 'memory', true);
```

Comment:
would the arguments of the tool call be very important for determining if the tool call should be trusted?

Reply:
Here's an example of where the tool is trusted, but the input is really the problem

https://github.com/microsoft/vscode/blob/aa8153f0041dcede6ac5c6308ec1c0e98fc4bdbb/src/vs/workbench/contrib/chat/electron-sandbox/tools/fetchPageTool.ts#L91-L118

Reply:
Yes. I anticipate we'll do something more advanced in the future / if we get more indicators of "risk" from MCP. This is just a first step.